src/components/views/settings/encryption/RecoveryPanel.tsx

```diff
@@ -105,33 +81,41 @@ export function RecoveryPanel({ onChangeRecoveryKeyClick }: RecoveryPanelProps):
                     label={_t("settings|encryption|recovery|title")}
                 />
             }
-            subHeading={<Subheader state={state} />}
+            subHeading={_t("settings|encryption|recovery|description")}
             data-testid="recoveryPanel"
         >
             {content}
         </SettingsSection>
     );
 }
 
-interface SubheaderProps {
+interface RecoveryPanelOutOfSyncProps {
     /**
-     * The state of the recovery panel.
+     * Callback for when the user has finished entering their recovery key.
      */
-    state: State;
+    onFinish: () => Promise<void>;
 }
 
 /**
- * The subheader for the recovery panel.
+ * This component is shown when the user secrets are not cached and the recovery key needs to be enter.
  */
-function Subheader({ state }: SubheaderProps): JSX.Element {
-    // If the secrets are not cached, we display a warning message.
-    if (state !== "secrets_not_cached") return <>{_t("settings|encryption|recovery|description")}</>;
-
+export function RecoveryPanelOutOfSync({ onFinish }: RecoveryPanelOutOfSyncProps): JSX.Element {
     return (
-        <SettingsSubheader
-            label={_t("settings|encryption|recovery|description")}
-            state="error"
-            stateMessage={_t("settings|encryption|recovery|key_storage_warning")}
-        />
+        <SettingsSection
+            legacy={false}
+            heading={_t("settings|encryption|recovery|title")}
+            subHeading={
+                <SettingsSubheader
+                    label={_t("settings|encryption|recovery|description")}
+                    state="error"
+                    stateMessage={_t("settings|encryption|recovery|key_storage_warning")}
+                />
+            }
+            data-testid="recoveryPanel"
+        >
+            <Button size="sm" kind="primary" Icon={KeyIcon} onClick={async () => await accessSecretStorage(onFinish)}>
```

Comment:
I don't think we should call `onFinish` from *within* `accessSecretStorage`; rather, we should wait for `accessSecretStorage` to complete successfully and *then* call `onFinish`.